sw/device/tests/entropy_src_kat_test.c

```diff
@@ -61,29 +106,55 @@ static void entropy_with_fw_override_enable(dif_entropy_src_t *entropy_src) {
  * See:
  * https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/secure-hashing
  */
-void test_sha384_kat(dif_entropy_src_t *entropy) {
-  CHECK_DIF_OK(dif_entropy_src_set_enabled(entropy, kDifToggleDisabled));
-  entropy_with_fw_override_enable(entropy);
-  CHECK_DIF_OK(dif_entropy_src_conditioner_start(entropy));
+
+void test_sha384_kat(dif_entropy_src_t *entropy_src) {
+  CHECK_DIF_OK(dif_entropy_src_set_enabled(entropy_src, kDifToggleDisabled));
+  entropy_with_fw_override_enable(entropy_src);
+
+  // Though most of the entropy_src state is cleared on disable, the
+  // SHA3 conditioner accumulates entropy even from aborted seeds. For
+  // the KAT though, we must clear any previously absorbed entropy.
+  flush_sha3_conditioner(entropy_src);
 
   const uint32_t kInputMsg[kEntropyFifoBufferSize] = {
       0xa52a0da9, 0xcae141b2, 0x6d5bab9d, 0x2c3e5cc0, 0x225afc93, 0x5d31a610,
       0x91b7f960, 0x0d566bb3, 0xef35e170, 0x94ba7d8e, 0x534eb741, 0x6b60b0da,
   };
-  CHECK_DIF_OK(dif_entropy_src_observe_fifo_write(entropy, kInputMsg,
-                                                  ARRAYSIZE(kInputMsg)));
 
-  CHECK_DIF_OK(dif_entropy_src_conditioner_stop(entropy));
+  CHECK_DIF_OK(dif_entropy_src_conditioner_start(entropy_src));
+
+  dif_result_t op_result;
+  uint32_t fail_count = 0;
+  uint32_t count;
+  uint32_t total = 0;
+
+  // Load the input data.
+  do {
+    op_result = dif_entropy_src_observe_fifo_write(
+        entropy_src, kInputMsg, ARRAYSIZE(kInputMsg), &count);
```

Comment:
Don't you need to compute write len and adjust the buffer index?
I mean what will happen if you are not able to send it all in the first iteration?

Reply:
Good catch!  You absolutely do.  Thanks @engdoreis!

Reply:
done